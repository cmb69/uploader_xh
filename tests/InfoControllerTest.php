<?php

namespace Uploader;

use ApprovalTests\Approvals;
use PHPUnit\Framework\TestCase;
use Plib\FakeSystemChecker;
use Plib\View;

class InfoControllerTest extends TestCase
{
    public function testDefaultActionShowsPluginInfo(): void
    {
        $sut = new InfoController(
            "./plugins/uploader/",
            new FakeSystemChecker(true),
            new View("./views/", XH_includeVar("./languages/en.php", "plugin_tx")["uploader"])
        );
        $response = $sut();
        $this->assertSame("Uploader 1.1-dev", $response->title());
        Approvals::verifyHtml($response->output());
    }
}
