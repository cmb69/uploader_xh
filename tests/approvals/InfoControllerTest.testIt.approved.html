
<h1>Uploader – 1.0beta2</h1>
<div>
  <h2>System check</h2>
  <p class="xh_success">Checking that PHP version ≥ 7.1.0 … okay</p>
  <p class="xh_success">Checking that the PHP extension 'json' is loaded … okay</p>
  <p class="xh_success">Checking that CMSimple_XH version ≥ 1.7.0 … okay</p>
  <p class="xh_success">Checking that the CMSimple_XH plugin 'jquery' is installed … okay</p>
  <p class="xh_success">Checking that '../uploader/config/' is writable … okay</p>
  <p class="xh_success">Checking that '../uploader/css/' is writable … okay</p>
  <p class="xh_success">Checking that '../uploader/languages/' is writable … okay</p>
</div>
